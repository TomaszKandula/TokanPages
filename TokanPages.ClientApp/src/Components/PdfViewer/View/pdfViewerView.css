.pdf-canvas {
    margin-left: auto;
    margin-right: auto;
    display: block;
    max-width: 928px;
    height: 100%;
    width: 100%;
}
