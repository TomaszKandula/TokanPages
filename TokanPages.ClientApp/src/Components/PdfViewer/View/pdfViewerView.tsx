import * as React from "react";
import { Card, CardContent, Container } from "@material-ui/core";
import NavigateBeforeIcon from "@material-ui/icons/NavigateBefore";
import NavigateNextIcon from "@material-ui/icons/NavigateNext";
import ReportProblemIcon from "@material-ui/icons/ReportProblem";
import DescriptionIcon from "@material-ui/icons/Description";
import { DownloadAsset, PdfCanvas, ProgressBar } from "../../../Shared/Components";

interface PdfViewerViewProps {
    isLoading: boolean;
    hasNoFilePrompt: boolean;
    hasError: boolean;
    content: {
        caption: string;
        text: string;
    };
    currentPage: number;
    numPages: number;
    pdfDocument: any;
    scale?: number;
    pdfUrl?: string;
    background?: string;
    onPreviousPage?: React.MouseEventHandler<SVGSVGElement>;
    onNextPage?: React.MouseEventHandler<SVGSVGElement>;
}

interface RenderIconOrErrorProps {
    isLoading: boolean;
    hasError: boolean;
    pdfUrl?: string;
}

const RenderIcon = (props: RenderIconOrErrorProps) => {
    return props.hasError ? <ReportProblemIcon /> : <DownloadAsset url={props.pdfUrl ?? ""} />;
};

const RenderIconOrLoading = (props: RenderIconOrErrorProps): React.ReactElement => {
    return props.isLoading && !props.hasError ? <ProgressBar size={20} /> : <RenderIcon {...props} />;
};

const RenderDocument = (props: PdfViewerViewProps): React.ReactElement => {
    return (
        <section className={`section ${props.background ?? ""}`}>
            <Container className="container-wide-1000">
                <div className="pt-80 pb-48">
                    <Card elevation={0} className="card">
                        <CardContent className="card-content">
                            <div className="pdf-header">
                                <div className="pdf-header-download-icon">
                                    <RenderIconOrLoading
                                        isLoading={props.isLoading}
                                        hasError={props.hasError}
                                        pdfUrl={props.pdfUrl}
                                    />
                                </div>
                                <div className="pdf-header-pages">
                                    {props.currentPage} / {props.numPages}
                                </div>
                                <div className="pdf-header-buttons-container">
                                    <NavigateBeforeIcon className="pdf-header-buttons" onClick={props.onPreviousPage} />
                                    <NavigateNextIcon className="pdf-header-buttons" onClick={props.onNextPage} />
                                </div>
                            </div>
                            <div className="pdf-canvas-wrapper">
                                <PdfCanvas
                                    pdfDocument={props.pdfDocument}
                                    pageNumber={props.currentPage}
                                    scale={props.scale ?? 1.5}
                                    htmlAttributes={{ className: "pdf-canvas" }}
                                />
                            </div>
                        </CardContent>
                    </Card>
                </div>
            </Container>
        </section>
    );
}

const RenderNoDocumentPrompt = (props: PdfViewerViewProps): React.ReactElement => {
    return (
        <section className={`section ${props.background ?? ""}`}>
            <Container className="container-wide">
                <div className="pt-80 pb-48">
                    <Card elevation={0} className="card">
                        <CardContent className="card-content">
                            <div className="pdf-nodocument-background mt-25 mb-25">
                                <div className="pdf-nodocument-icon-holder vertical-centre">
                                    <DescriptionIcon className="pdf-nodocument-icon vertical-centre" />
                                </div>
                            </div>
                            <h2 className="pdf-nodocument-caption text-centre mb-32">
                                {props.content.caption}
                            </h2>
                            <div className="pdf-nodocument-text text-centre mb-32">
                                {props.content.text}
                            </div>
                        </CardContent>
                    </Card>
                </div>
            </Container>
        </section>
    );
}

export const PdfViewerView = (props: PdfViewerViewProps): React.ReactElement => {
    return props.hasNoFilePrompt ? <RenderNoDocumentPrompt {...props} /> : <RenderDocument {...props} />;
};
