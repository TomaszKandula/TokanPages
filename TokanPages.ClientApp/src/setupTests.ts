// jest-dom adds custom jest matchers for asserting on DOM nodes.
// allows you to do things like:
// expect(element).toHaveTextContent(/react/i)
// learn more: https://github.com/testing-library/jest-dom
import "@testing-library/jest-dom";
import Enzyme from "enzyme";
import Adapter from "enzyme-adapter-react-16";
import enableHooks from "jest-react-hooks-shallow";

enableHooks(jest);

jest.mock("react-redux", () => (
{
    useSelector: jest.fn(),
    useDispatch: jest.fn()
}));

jest.mock("connected-react-router", () => (
{
    connectRouter: jest.fn(),
    routerMiddleware: jest.fn()
}));

Enzyme.configure({ adapter: new Adapter(), disableLifecycleMethods: false });

// environmental variables used during all tests
process.env.REACT_APP_API_VER="1";
process.env.REACT_APP_BACKEND="http://localhost:5000";
