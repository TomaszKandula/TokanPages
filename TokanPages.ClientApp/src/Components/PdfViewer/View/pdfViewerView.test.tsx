import "../../../setupTests";
import React from "react";
import { BrowserRouter } from "react-router-dom";
import { render } from "@testing-library/react";
import { PdfViewerView } from "./pdfViewerView";

describe("test component: pdfViewerView", () => {
    it("should render correctly '<PdfViewerView />' when content is loaded.", () => {
        const html = render(
            <BrowserRouter>
                <PdfViewerView
                    isLoading={false}
                    hasError={false}
                    currentPage={1}
                    numPages={1}
                    pdfDocument={{}}
                    scale={undefined}
                    pdfUrl="https://test.com/documents/some_document.pdf"
                    onPreviousPage={jest.fn()}
                    onNextPage={jest.fn()}
                    background={{ backgroundColor: "white" }}
                />
            </BrowserRouter>
        );

        expect(html).toMatchSnapshot();
    });
});
