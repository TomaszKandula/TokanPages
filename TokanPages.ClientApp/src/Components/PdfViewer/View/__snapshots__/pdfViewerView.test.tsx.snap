// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test component: pdfViewerView should render correctly '<PdfViewerView />' when content is loaded. 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <section
        class="makeStyles-section-1"
        style="background-color: white;"
      >
        <div
          class="MuiContainer-root makeStyles-container-2 MuiContainer-maxWidthLg"
        >
          <div
            class="MuiBox-root MuiBox-root-8"
          >
            <div
              class="MuiGrid-root MuiGrid-container MuiGrid-direction-xs-column MuiGrid-justify-content-xs-center"
            >
              <div
                class="MuiBox-root MuiBox-root-9 makeStyles-header-3"
              >
                <a
                  href="https://test.com/documents/some_document.pdf&canDownload=true"
                  style="color: rgb(0, 0, 0);"
                >
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M19 9h-4V3H9v6H5l7 7 7-7zM5 18v2h14v-2H5z"
                    />
                  </svg>
                </a>
                <div
                  class="makeStyles-header_pages-4"
                >
                  1
                   / 
                  1
                </div>
                <div>
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root makeStyles-header_buttons-5"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M15.41 7.41L14 6l-6 6 6 6 1.41-1.41L10.83 12z"
                    />
                  </svg>
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root makeStyles-header_buttons-5"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M10 6L8.59 7.41 13.17 12l-4.58 4.59L10 18l6-6z"
                    />
                  </svg>
                </div>
              </div>
              <div
                class="MuiBox-root MuiBox-root-10 makeStyles-canvasWrapper-7"
              >
                <canvas
                  class="makeStyles-canvas-6"
                />
              </div>
            </div>
          </div>
        </div>
      </section>
    </div>
  </body>,
  "container": <div>
    <section
      class="makeStyles-section-1"
      style="background-color: white;"
    >
      <div
        class="MuiContainer-root makeStyles-container-2 MuiContainer-maxWidthLg"
      >
        <div
          class="MuiBox-root MuiBox-root-8"
        >
          <div
            class="MuiGrid-root MuiGrid-container MuiGrid-direction-xs-column MuiGrid-justify-content-xs-center"
          >
            <div
              class="MuiBox-root MuiBox-root-9 makeStyles-header-3"
            >
              <a
                href="https://test.com/documents/some_document.pdf&canDownload=true"
                style="color: rgb(0, 0, 0);"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M19 9h-4V3H9v6H5l7 7 7-7zM5 18v2h14v-2H5z"
                  />
                </svg>
              </a>
              <div
                class="makeStyles-header_pages-4"
              >
                1
                 / 
                1
              </div>
              <div>
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root makeStyles-header_buttons-5"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M15.41 7.41L14 6l-6 6 6 6 1.41-1.41L10.83 12z"
                  />
                </svg>
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root makeStyles-header_buttons-5"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M10 6L8.59 7.41 13.17 12l-4.58 4.59L10 18l6-6z"
                  />
                </svg>
              </div>
            </div>
            <div
              class="MuiBox-root MuiBox-root-10 makeStyles-canvasWrapper-7"
            >
              <canvas
                class="makeStyles-canvas-6"
              />
            </div>
          </div>
        </div>
      </div>
    </section>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
